### PR TITLE
Removed provenance from the reward calculation

### LINKED ISSUE #2744: Remove reward provenance
The ledger events are replacing the reward provenance, and hence we should remove reward provenance.

### PR DESCRIPTION
**Pull request by TimSheard**
Removed provenance from the reward calculation, and also all the tests. This involved all the reward-pulsing tests (that used to involve provenance, and now don't) and also the tests that provenance didn't affect the underlying computation.

closes #2744

### CODE REVIEW

File: eras/shelley/impl/src/Cardano/Ledger/Shelley/API/Wallet.hs [hunk 1]
```diff
@@ -430,12 +427,10 @@ getRewardProvenance ::
   ) =>
   Globals ->
   NewEpochState era ->
-  (RewardUpdate (Crypto era), RewardProvenance (Crypto era))
+  (RewardUpdate (Crypto era))
 getRewardProvenance globals newepochstate =
```
Comment (JaredCorduan) [inline eras/shelley/impl/src/Cardano/Ledger/Shelley/API/Wallet.hs:433]: This is great @TimSheard , thanks for doing this clean up!  Before we merge this, we should make sure we coordinate with the consensus team about how best to deprecate their [query](https://github.com/input-output-hk/ouroboros-network/blob/a10ba425a5db0b7a009d6f16f066bb70b2bd3707/ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs#L250-L251). We've already gutted most of the information from that query, so it's really just about deprecating the query in the most clear and helpful way we can.

File: eras/shelley/impl/cardano-ledger-shelley.cabal [hunk 2]
```diff
@@ -55,7 +55,7 @@ library
     Cardano.Ledger.Shelley.PoolRank
     Cardano.Ledger.Shelley.PParams
     Cardano.Ledger.Shelley.Rewards
-    Cardano.Ledger.Shelley.RewardProvenance
+    Cardano.Ledger.Shelley.RewardProvenance
```
Comment (JaredCorduan) [inline eras/shelley/impl/cardano-ledger-shelley.cabal:58]: trailing whitespace (I _think_ ormolu will notice as well)

**Review by JaredCorduan — APPROVED** [review]
looks great, thank you!

File: eras/shelley/impl/cardano-ledger-shelley.cabal (hunk 2 above)
Comment (lehins) [reply eras/shelley/impl/cardano-ledger-shelley.cabal:58]: Ormolu does not touch cabal files

**Comment by JaredCorduan** [comment]
`ExtraneousScriptWitnessesUTXOW` in the generation strikes again. https://github.com/input-output-hk/cardano-ledger/issues/2811

I'm going to merge this PR, as this failure is a known in the generators and is unrelated.

**Pull request merged**